package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,228 @@
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        # for transform function
+        self.start = kwargs.get('start')
+        self.stop = kwargs.get('stop')
+        self.step = kwargs.get('step')
+
+        if demean_ref:
+            self._demean = True
+            self._reference = demean_ref
+        else:
+            self._demean = False
+
+        self._atoms = atomgroup.select_atoms(select)
+        self.n_components = n_components
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        if self._demean:
+            self._ref_atoms = self._reference
+            self._ref_cog = self._ref_atoms.center_of_geometry()
+
+    def _single_frame(self):
+        if self._demean:
+            # TODO make this more functional, initial proof of concept
+            mobile_atoms, old_rmsd = _fit_to(self._atoms.positions,
+                                             self._ref_atoms.positions,
+                                             self._atoms, self._ref_cog,
+                                             self._atoms.center_of_geometry())
+            # now all structures are aligned to reference
+            x = mobile_atoms.positions.ravel()
+        else:
+            x = self._atoms.positions.ravel()
+        self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
+        e_vals, e_vects = np.linalg.eig(self.cov)
+        sort_idx = np.argsort(e_vals)[::-1]
+        self.variance = e_vals[sort_idx]
+        self.p_components = e_vects[sort_idx]
+        self.cumulated_variance = (np.cumsum(self.variance) /
+                                   np.sum(self.variance))
+        self._calculated = True
+    def transform(self, atomgroup, n_components=None):
+        """Apply the dimensionality reduction on a trajectory
+
+        Parameters
+        ----------
+        n_components: int, optional
+            The number of components to be projected onto, default maps onto
+            all components.
+        Returns
+        -------
+        pca_space : array, shape (number of frames, number of components)
+        """
+        if not self._calculated:
+            self.run()
+        # TODO has to accept universe or trajectory slice here
+        if self._atoms != atomgroup:
+            warnings.warn('This is a transform for different atom types.')
+
+        dot = np.zeros((self._u.trajectory.n_frames, atomgroup.n_atoms*3))
+
+        for i, ts in enumerate(self._u.trajectory[self.start:self.stop:self.step]):
```

Comment:
So when you call transform. You already have a PCA matrix. This means a simple rotation matrix. We can apply this to any atomgroup that has a compatible number of atoms. This is what transform should do. We do not use the coordinates of the atomgroup we used to calculate the PCA but the ones of the atomgroup of the which we want to transform into the PCA space. I hope that helps.